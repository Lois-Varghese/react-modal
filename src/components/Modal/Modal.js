import React, {useRef, useEffect} from 'react';
import Button from '../../common/Button/Button';
import CloseIcon from '../../common/CloseIcon/CloseIcon';
import './Modal.scss';

const Modal = ({modalStyle, children, show, onClose, backdropStyle}) => {
  const modalRef = useRef(null);
  useEffect(() => {
    if (show) {
      modalRef.current.classList.add('visible');
    } else {
      modalRef.current.classList.remove('visible');
    }
  }, [show]);

  return (
    <>
      <div
        ref={modalRef}
        style={backdropStyle}
        onClick={onClose}
        className="modal__wrap">
        <div
          style={modalStyle}
          onClick={(e) => e.stopPropagation()}
          className="modal">
          <Button
            onClick={onClose}
            style={{
              width: 30,
              height: 30,
              position: 'absolute',
              top: 0,
              right: 0,
              margin: '12px',
            }}
            className="close__btn">
            <CloseIcon height="26px" width="26px" className="close__icon" />
          </Button>
          {children}
        </div>
      </div>
    </>
  );
};

export default Modal;
