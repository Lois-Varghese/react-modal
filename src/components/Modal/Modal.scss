@import '../../styles/abstracts/variables';

.modal__wrap {
  position: fixed;
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 100;
  overflow-x: hidden;
  pointer-events: none;
  opacity: 0;
  transition: opacity 250ms 700ms ease;
}

.visible {
  pointer-events: auto;
  opacity: 1;
  transition: all 300ms ease-in-out;
}

.modal {
  overflow-y: scroll;
  overflow-x: hidden;
  position: relative;
  display: block;
  min-height: 400px;
  min-width: 400px;
  margin: 0 auto;
  margin-top: 20px;
  margin-bottom: 20px;
  border-radius: 4px;
  padding-bottom: 20px;
  background-color: $color_white;
  align-self: center;
  box-shadow: 0 12px 25px 0 $color_black;
  opacity: 0;
  transform: scale(0.6);
  transition: opacity 250ms 250ms ease, transform 300ms 250ms ease;
  transform: scale(0);
}

.visible .modal {
  opacity: 1;
  transform: scale(1);
  transition: opacity 250ms 500ms ease, transform 350ms 500ms ease;
}

.close__btn {
  background-color: $color_grey;
  border-radius: 24px;
  border: none;
  display: flex;
  fill: $color_white;
  justify-content: center;
  align-items: center;
}
