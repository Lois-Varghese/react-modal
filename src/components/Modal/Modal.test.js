import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Modal from "./Modal";

configure({ adapter: new Adapter() });
const wrapper = shallow(<Modal />);
let container, containerProp;

describe("Render component", () => {
  it("should render Modal Component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe("render first <div>", () => {
  beforeEach(() => {
    container = wrapper.find("div").at(0);
    containerProp = container.props();
  });

  it("should have a <div> with className", () => {
    expect(containerProp.className).toEqual("modal__wrap");
  });
});

describe("render second <div>", () => {
  beforeEach(() => {
    container = wrapper.find("div").at(1);
    containerProp = container.props();
  });

  it("should have a <div> with className", () => {
    expect(containerProp.className).toEqual("modal");
  });
});
