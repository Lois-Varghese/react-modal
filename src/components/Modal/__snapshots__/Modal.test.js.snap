// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render component should render Modal Component 1`] = `
<Fragment>
  <div
    className="modal__wrap"
  >
    <div
      className="modal"
      onClick={[Function]}
    >
      <Button
        className="close__btn"
        style={
          Object {
            "height": 30,
            "margin": "12px",
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 30,
          }
        }
      >
        <CloseIcon
          className="close__icon"
          height="26px"
          width="26px"
        />
      </Button>
    </div>
  </div>
</Fragment>
`;
